src/k2/tso/service/TSOController.cpp

```diff
@@ -294,9 +298,12 @@ seastar::future<> TSOService::TSOController::DoHeartBeat()
                 _myLease = std::get<0>(newLeaseAndThreshold);
                 _controlInfoToSend.ReservedTimeShreshold = std::get<1>(newLeaseAndThreshold);
 
-                uint64_t newCurTimeTSECount = TimeAuthorityNow();
-                K2ASSERT(log::tsoserver, _controlInfoToSend.ReservedTimeShreshold > newCurTimeTSECount && _myLease > newCurTimeTSECount,
-                    "new lease and ReservedTimeThreshold should be in the future.");
+                if (!_controlInfoToSend.IgnoreThreshold)
+                {
+                    uint64_t newCurTimeTSECount = TimeAuthorityNow();
+                    K2ASSERT(log::tsoserver, _controlInfoToSend.ReservedTimeShreshold > newCurTimeTSECount && _myLease > newCurTimeTSECount,

```

Comment:
This assert is what currently still causing TSO fail to issue timestamp on dev box.